File: system-test/java/src/main/java/com/optivem/shop/testkit/driver/adapter/external/tax/client/TaxStubClient.java
package com.optivem.shop.testkit.driver.adapter.external.tax.client;

import com.optivem.shop.testkit.driver.adapter.shared.client.http.HttpStatus;
import com.optivem.shop.testkit.driver.adapter.external.tax.client.dtos.ExtGetCountryResponse;
import com.optivem.shop.testkit.driver.adapter.external.tax.client.dtos.error.ExtTaxErrorResponse;
import com.optivem.shop.testkit.common.Result;
import com.optivem.shop.testkit.driver.adapter.shared.client.wiremock.JsonWireMockClient;

public class TaxStubClient extends BaseTaxClient {
    private static final String WIREMOCK_COUNTRIES_ENDPOINT = "/tax/api/countries/";

    private final JsonWireMockClient wireMockClient;

    public TaxStubClient(String baseUrl) {
        super(baseUrl);
        this.wireMockClient = new JsonWireMockClient(baseUrl);
    }

    public Result<Void, ExtTaxErrorResponse> configureGetCountry(String country, ExtGetCountryResponse response) {
        return wireMockClient.stubGet(WIREMOCK_COUNTRIES_ENDPOINT + country, HttpStatus.OK, response)
                .mapError(ExtTaxErrorResponse::new);
    }

    public void removeStubs() {
        wireMockClient.removeStubs();
    }
}

File: system-test/typescript/src/testkit/driver/adapter/external/tax/client/TaxStubClient.ts
import { Result, success } from '../../../../../common/result.js';
import type { TaxErrorResponse } from '../../../../port/external/tax/dtos/TaxErrorResponse.js';
import type { ReturnsTaxRateRequest } from '../../../../port/external/tax/dtos/ReturnsTaxRateRequest.js';
import { JsonWireMockClient } from '../../../shared/wiremock/wiremock-client.js';
import { BaseTaxClient } from './BaseTaxClient.js';

export class TaxStubClient extends BaseTaxClient {
  private readonly wireMock: JsonWireMockClient;

  constructor(baseUrl: string) {
    super(baseUrl);
    this.wireMock = new JsonWireMockClient(baseUrl);
  }

  async configureTaxRate(request: ReturnsTaxRateRequest): Promise<Result<void, TaxErrorResponse>> {
    await this.wireMock.stubGet(`/tax/api/countries/${request.country}`, {
      id: request.country,
      countryName: request.country,
      taxRate: parseFloat(request.taxRate),
    });
    return success(undefined);
  }

  async close(): Promise<void> {
    await this.wireMock.removeStubs();
  }
}
